go/libraries/doltcore/env/repo_state.go

```diff
@@ -44,8 +45,13 @@ type RepoStateWriter interface {
 type DocsReadWriter interface {
 	GetAllValidDocDetails() ([]doltdb.DocDetails, error)
 	PutDocsToWorking(ctx context.Context, docDetails []doltdb.DocDetails) error
+	PutDocsToStaged(ctx context.Context, docDetails []doltdb.DocDetails) (*doltdb.RootValue, error)
 	ResetWorkingDocsToStagedDocs(ctx context.Context) error
 	GetDocDetail(docName string) (doc doltdb.DocDetails, err error)
+	UpdateFSDocsToRootDocs(ctx context.Context, root *doltdb.RootValue, docs Docs) error
+	GetDocsWithNewerTextFromRoot(ctx context.Context, root *doltdb.RootValue, docs Docs) (Docs, error)
+	GetFS() filesys.Filesys
+	GetDocs() Docs
```

Comment:
This seems to be redundant with GetAllValidDocDetails, but actually does something completely different. Yikes.